MediaWikiKit/MediaWikiKit/MWKSavedPageList.m

```diff
@@ -106,12 +117,39 @@ - (void)performSaveWithCompletion:(dispatch_block_t)completion error:(WMFErrorHa
     }
 }
 
+#pragma mark - Schema Migration
+
++ (NSArray<NSDictionary*>*)savedEntryDataFromExportedData:(NSDictionary*)savedPageListData {
+    NSNumber* schemaVersionValue                = savedPageListData[MWKSavedPageExportedSchemaVersionKey];
+    MWKSavedPageListSchemaVersion schemaVersion = MWKSavedPageListSchemaVersionUnknown;
+    if (schemaVersionValue) {
+        schemaVersion = schemaVersionValue.unsignedIntegerValue;
+    }
+    switch (schemaVersion) {
+        case MWKSavedPageListSchemaVersionCurrent:
+            return savedPageListData[MWKSavedPageExportedEntriesKey];
+        case MWKSavedPageListSchemaVersionUnknown:
+            return [MWKSavedPageList savedEntryDataFromListWithUnknownSchema:savedPageListData];
+    }
+}
+
++ (NSArray<NSDictionary*>*)savedEntryDataFromListWithUnknownSchema:(NSDictionary*)data {
+    return [data[MWKSavedPageExportedEntriesKey] wmf_reverseArray];
+}
+
 #pragma mark - Export
 
-- (NSArray*)dataExport {
-    return [self.entries bk_map:^id (MWKSavedPageEntry* obj) {
-        return [obj dataExport];
+- (NSArray<NSDictionary*>*)exportedEntries {
+    return [self.entries bk_map:^NSDictionary*(MWKSavedPageEntry* entry) {
+        return [entry dataExport];
     }];
 }
 
+- (NSDictionary*)dataExport {
```

Comment:
i know, i _almost_ did it for generic `MWKList` serialization. then i stepped back and remembered this is all going away